### ISSUE TITLE
2016.11.6 - TCP Transport gives Exception

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#41778]
### Description of Issue/Question
In 2016.11.6 when running commands using TCP it gives `Exception ValueError`
```
Exception ValueError: 'I/O operation on closed file' in <bound method SyncWrapper.__del__ of <salt.utils.async.SyncWrapper object at 0x7f359f2c7d50>> ignored
```
### Setup
Install 2016.11.6, set ` transport: tcp`, accept key, and run commands.

### Steps to Reproduce Issue
```
root@localhost:~# salt \* test.ping
Exception ValueError: 'I/O operation on closed file' in <bound method SyncWrapper.__del__ of <salt.utils.async.SyncWrapper object at 0x7f59ca9ced50>> ignored
ubuntu.members.linode.com:
    True
root@localhost:~# 
```

### Versions Report
```
root@localhost:~# salt --versions
Salt Version:
           Salt: 2016.11.6
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.9.15-x86_64-linode81
         system: Linux
        version: Ubuntu 16.04 denial
```

**Comment by frogunder** [comment]
```
892c6d4d24684bb2dc8b5055304c923ce3e2f234 is the first bad commit
commit 892c6d4d24684bb2dc8b5055304c923ce3e2f234
Author: Sergey Kizunov <sergey.kizunov@ni.com>
Date:   Thu May 25 13:30:19 2017 -0500
```

@cachedout @skizunov

**Comment by skizunov** [comment]
@frogunder, @cachedout : I posted a fix for this in PR #41787
